### DISCUSSION TITLE
Creating models from existing database

### DISCUSSION THREAD

**Discussion by thetutlage** [adonisjs/core#1356] [Ideas]
**I receive a lot of requests asking to create migrations from models, just like the way Django does. I have a different take on this and sharing it here**

Fundamentally, migrations and models are de-coupled in both AdonisJS and Django. For example: If you have an existing database, you can just create models and start using it without creating any migrations.

But Django does provide extra tooling to create migrations from the existing models, so basically you save some time in writing them by hand. I personally avoided it for couple of reasons.

- Once you are driving your database from your models, then you have to define everything within your models. For example: The `data type (varchar)`, `null constraints`, `PK`, `FK` or maybe even `database triggers`. I think this makes models fat, as you don't need half of this information at runtime (after migrations are done).
- Very mature tooling is required to pull it off. Something, I don't have time for right now.

But, I do believe in the idea of automating things with migrations and models. So I decided to later work on something that can create/update models from the existing database tables.

- You have a existing database, you run `node ace make:models` and it creates all the models for you.
- The database can be a old database or something you created using migrations.
- If you make changes to the table, you can run `node ace update:model` and it will reflect the changes without messing up with your custom code in that file.

This approach will keep models simple (not configuring DB via models) + you are not defining columns by hand in the models and they are always up to date.

Again this also requires some effort, so de-prioritized over critical features

**Comment by Uttam-EWW** [comment]
`node ace make:models` & `node ace update:model` is not working. Is there something i'm missing?

**Reply by mastermunj** [reply]
@Uttam-EWW These commands won't work because they are not created. The post was to clarify that having these features requires some effort which aren't important considering other critical features require attention as of now.

**Reply by Uttam-EWW** [reply]
> 

Oh, okay. Got it. thank you.

**Comment by ricar2ce** [comment]
> But, I do believe in the idea of automating things with migrations and models. So I decided to later work on something that can create/update models from the existing database tables.
> 
> * You have a existing database, you run `node ace make:models` and it creates all the models for you.
> * The database can be a old database or something you created using migrations.
> * If you make changes to the table, you can run `node ace update:model` and it will reflect the changes without messing up with your custom code in that file.
> 
> This approach will keep models simple (not configuring DB via models) + you are not defining columns by hand in the models and they are always up to date.
> 
> Again this also requires some effort, so de-prioritized over critical features

Has anything been done on this topic?

I have a complete database in postgres and I would like to make a new api in Adonis, but I need not to have to define each model by hand, any suggestions please

**Comment by mattstrayer** [comment]
Also have a need for this feature and would make my migration from Django to Adonis much easier!

**Reply by thetutlage** [reply]
Need contributors for it.

**Comment by stereobooster** [comment]
Related https://github.com/adonisjs/core/discussions/4017

**Comment by letiesperon** [comment]
Would love to see this feature

**Comment by mathhas-dev** [comment]
Any update?

**Comment by orafal-dev** [comment]
Is it still considered to be done?

**Comment by MrSharpp** [comment]
any updates?

**Comment by DilwoarH** [comment]
@thetutlage - are you still considering this?

**Comment by IcyFoxe** [comment]
Would also be interesting about this kind of functionality to keep models and tables in sync.

**Reply by thetutlage** [reply]
Would love to have someone champion this proposal. I personally haven't been able to find time for it.
